src/languages/Go.ts

```diff
@@ -0,0 +1,31 @@
+import Parser from 'tree-sitter';
+import TreeSitterGo from 'tree-sitter-go';
+import { Declaration } from '../types';
+
+const extractTypeDeclarationName = (node: Parser.SyntaxNode): string => {
+    return node.text.replace('type ', '').split(' ')[0];
+};
+
+const extractFunctionDeclarationName = (node: Parser.SyntaxNode): string => {
+    return node.text.replace('func ', '').split(' ')[0];
+};
```

Comment:
`func (h *handler) getExapmpleByID(id int)`
would return `(h *handler)`

but we can fix that in another PR

Reply:
That should be fixed once we move to tree-sitter fields.